### PR TITLE
Updated some design and inspection code

### PR DESCRIPTION
**Pull request by EBrown8534**
Originally inspections were integrated into the `HomeController`, so I extracted an `InspectionsController` and moved all that out there. They also dynamically generate from the `Rubberduck.dll` file, and also fixed the `Ninject` (with the help of @retailcoder) so that it will allow us to load all Inspections. This allows us to just update the Rubberduck.dll and the new inspections (or removed inspections) are adjusted automatically.

### CODE REVIEW

File: RubberduckWeb/RubberduckWeb/App_Start/NinjectWebCommon.cs [hunk 1]
```diff
@@ -93,29 +104,19 @@ private static void ApplyDefaultInterfacesConvention(IKernel kernel, IEnumerable
             kernel.Bind(t => t.From(assemblies)
                 .SelectAllClasses()
                 // inspections & factories have their own binding rules
-                .Where(type => !type.Name.EndsWith("Factory") && !type.Name.EndsWith("ConfigProvider") && !type.GetInterfaces().Contains(typeof(IInspection)))
+                .Where(type => !type.Name.Contains("VBA") && !type.Name.EndsWith("Factory") && !type.Name.EndsWith("ConfigProvider") && !type.GetInterfaces().Contains(typeof(IInspection)))
```
Comment (retailcoder) [inline RubberduckWeb/RubberduckWeb/App_Start/NinjectWebCommon.cs]: not sure `!type.Name.Contains("VBA")` does what you think it does...

Comment (EBrown8534) [inline RubberduckWeb/RubberduckWeb/App_Start/NinjectWebCommon.cs]: Probably not, that's an artifact that can disappear now.

File: RubberduckWeb/RubberduckWeb/Controllers/InspectionsController.cs [hunk 2]
```diff
@@ -53,7 +80,72 @@ public Task<PartialViewResult> GetInspectionResults(string code)
                         && (ir.QualifiedSelection.QualifiedName.Name == "TestProject1." || ir.QualifiedSelection.QualifiedName.Name == "TestProject1.TestModule1")
                     );
 
-            return Task.FromResult(PartialView("~/Views/Home/InspectionResults.cshtml", results));
+            return Task.FromResult(PartialView("InspectionResults", results));
+        }
+
+        public static string FormatInspectionName(IInspection inspection)
+        {
+            return InsertOnCharacter(inspection.Name.Remove(inspection.Name.Length - 10), CharacterType.UpperLetter, " ");
```
Comment (retailcoder) [inline RubberduckWeb/RubberduckWeb/Controllers/InspectionsController.cs]: what's the `-10`?

File: RubberduckWeb/RubberduckWeb/App_Start/NinjectWebCommon.cs [hunk 3]
```diff
@@ -93,29 +104,19 @@ private static void ApplyDefaultInterfacesConvention(IKernel kernel, IEnumerable
...
 
             // skip inspections with undesirable dependencies
             foreach (var inspection in inspections)
             {
-                if (inspection.Name == nameof(ImplicitActiveSheetReferenceInspection) ||
-                    inspection.Name == nameof(ImplicitActiveWorkbookReferenceInspection) ||
-                    inspection.Name == nameof(ParameterNotUsedInspection) ||
-                    inspection.Name == nameof(EncapsulatePublicFieldInspection) ||
-                    inspection.Name == nameof(UseMeaningfulNameInspection) ||
-                    inspection.Name == nameof(UndeclaredVariableInspection))
```
Comment (EBrown8534) [inline RubberduckWeb/RubberduckWeb/App_Start/NinjectWebCommon.cs]: I'll disable that in the actual `DefaultInspector`.

File: RubberduckWeb/RubberduckWeb/Controllers/InspectionsController.cs (hunk 2 above)
Comment (EBrown8534) [inline RubberduckWeb/RubberduckWeb/Controllers/InspectionsController.cs]: A magic number for `"Inspection".Length`. Fixing that. ;)

**Pull request merged**